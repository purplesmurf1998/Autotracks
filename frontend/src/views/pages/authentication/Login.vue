<template>
  <div class="c-app flex-row align-items-center" :class="{ 'c-dark-theme': $store.state.darkMode }">
    <CContainer>
      <CRow class="justify-content-center">
        <CCol md="8">
          <CCardGroup>
            <CCard class="p-4">
              <CCardBody>
                <CForm @submit.prevent="submitForm">
                  <h1>Login</h1>
                  <p class="text-muted">Sign In to your account</p>
                  <CInput
                    placeholder="Email"
                    autocomplete="username email"
                    v-model="email"
                    prepend="@"
                  />
                  <CInput
                    placeholder="Password"
                    type="password"
                    autocomplete="curent-password"
                    v-model="password"
                  >
                    <template #prepend-content><CIcon name="cil-lock-locked"/></template>
                  </CInput>
                  <CRow>
                    <CCol col="6" class="text-left">
                      <CButton color="primary" class="px-4" type="submit">Login</CButton>
                    </CCol>
                    <CCol col="6" class="text-right">
                      <CButton color="link" class="px-0">Forgot password?</CButton>
                      <CButton color="link" class="d-lg-none">Register now!</CButton>
                    </CCol>
                  </CRow>
                </CForm>

                <CAlert show color="danger" v-if="errorMsg" class="mt-2">Invalid Credentials</CAlert>
              </CCardBody>
            </CCard>
            <CCard
              color="primary"
              text-color="white"
              class="text-center py-5 d-md-down-none"
              body-wrapper
            >
              <CCardBody>
                <h2>Sign up</h2>

                <p>Sign up for an admin account and start tracking your inventory today! Get three users FREE when signing up for one year.</p>
                <router-link to="/pages/register">
                  <CButton
                    color="light"
                    variant="outline"
                    size="lg"
                  >
                    Register Now!
                  </CButton>
                </router-link>
              </CCardBody>
            </CCard>
          </CCardGroup>
        </CCol>
      </CRow>
    </CContainer>
  </div>
</template>

<script>
export default {
  name: 'Login',
  data() {
    return {
      email: '',
      password: '',
      errorMsg: null
    }
  },
  methods: {
    async submitForm () {
      if (this.email == '' || this.password == '') {
        this.showError('Invalid credentials');
      } else {
        // try to log the user in using the vuex store
        const response = await this.$store.dispatch('login', {
          email: this.email,
          password: this.password
        });
        
        // if not successful, show the error message
        if (!response.success) {
          this.showError(response.message);
        } 
        // if successful, redirect the user to the dashboad
        else {
          this.$router.push("/dashboard");
        }
      }
    },
    showError (errorMsg) {
      this.password = '';
      this.errorMsg = errorMsg;
      setTimeout(() => {
        this.errorMsg = null;
      }, 4000)

    },
  }
}
</script>
