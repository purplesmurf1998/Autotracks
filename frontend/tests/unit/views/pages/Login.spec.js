import {createLocalVue, shallowMount} from '@vue/test-utils'
import CoreuiVue from '@coreui/vue-pro'
import Vuex from 'vuex'

import Login from './../../../../src/views/pages/authentication/Login.vue'

const localVue = createLocalVue()
localVue.use(CoreuiVue)
localVue.use(Vuex)

const store = new Vuex.Store({
  state: {
    sidebarShow: 'responsive',
    sidebarMinimize: false,
    asideShow: false,
    darkMode: false
  }
})

describe('Login.vue', () => {
  it('has a name', () => {
    expect(Login.name).toBe('Login')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Login, { store, localVue })
    expect(wrapper.vm).toBeTruthy()
  })
  it('is Login', () => {
    const wrapper = shallowMount(Login, { store, localVue })
    expect(wrapper.findComponent(Login)).toBeTruthy()
  })
  it('should render correct content', () => {
    const wrapper = shallowMount(Login, { store, localVue })
    expect(wrapper.find('h1').text()).toMatch('Login')
    expect(wrapper.find('p').text()).toMatch('Sign In to your account')
    expect(wrapper.find('h2').text()).toMatch('Sign up')
    // expect(wrapper.find('p').text()).toMatch('Sign up for an admin account and start tracking your inventory today! Get three users FREE when signing up for one year')
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Login, { store, localVue })
    expect(wrapper.element).toMatchSnapshot()
  })
})
