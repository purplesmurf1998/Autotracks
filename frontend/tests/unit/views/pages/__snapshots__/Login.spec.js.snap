// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login.vue renders correctly 1`] = `
<div
  class="c-app flex-row align-items-center"
>
  <ccontainer-stub
    tag="div"
  >
    <crow-stub
      class="justify-content-center"
      gutters="true"
      tag="div"
    >
      <ccol-stub
        md="8"
        tag="div"
      >
        <ccardgroup-stub
          tag="div"
        >
          <ccard-stub
            class="p-4"
          >
            <ccardbody-stub>
              <cform-stub>
                <h1>
                  Login
                </h1>
                 
                <p
                  class="text-muted"
                >
                  Sign In to your account
                </p>
                 
                <cinput-stub
                  autocomplete="username email"
                  id="email-txt-field"
                  lazy="400"
                  placeholder="Email"
                  type="text"
                  value=""
                />
                 
                <cinput-stub
                  autocomplete="curent-password"
                  id="password-txt-field"
                  lazy="400"
                  placeholder="Password"
                  type="password"
                  value=""
                />
                 
                <crow-stub
                  gutters="true"
                  tag="div"
                >
                  <ccol-stub
                    class="text-left"
                    col="6"
                    tag="div"
                  >
                    <cbutton-stub
                      activeclass="router-link-active"
                      class="px-4"
                      color="primary"
                      event="click"
                      exactactiveclass="router-link-exact-active"
                      id="login-btn"
                      routertag="a"
                      target="_self"
                      type="submit"
                    >
                      Login
                    </cbutton-stub>
                  </ccol-stub>
                   
                  <ccol-stub
                    class="text-right"
                    col="6"
                    tag="div"
                  >
                    <cbutton-stub
                      activeclass="router-link-active"
                      class="px-0"
                      color="link"
                      event="click"
                      exactactiveclass="router-link-exact-active"
                      id="forgot-pswrd-btn"
                      routertag="a"
                      target="_self"
                      type="button"
                    >
                      Forgot password?
                    </cbutton-stub>
                  </ccol-stub>
                </crow-stub>
                 
                <!---->
              </cform-stub>
            </ccardbody-stub>
          </ccard-stub>
           
          <ccard-stub
            bodywrapper="true"
            class="text-center py-5 d-md-down-none"
            color="primary"
            textcolor="white"
          >
            <ccardbody-stub>
              <h2>
                Sign up
              </h2>
               
              <p>
                Sign up for an admin account and start tracking your inventory today! Get three users FREE when signing up for one year.
              </p>
               
              <router-link
                to="/pages/register"
              >
                <cbutton-stub
                  activeclass="router-link-active"
                  color="light"
                  event="click"
                  exactactiveclass="router-link-exact-active"
                  routertag="a"
                  size="lg"
                  target="_self"
                  type="button"
                  variant="outline"
                >
                  
                  Register Now!
                
                </cbutton-stub>
              </router-link>
            </ccardbody-stub>
          </ccard-stub>
        </ccardgroup-stub>
      </ccol-stub>
    </crow-stub>
  </ccontainer-stub>
</div>
`;
